scrapy/core/spidermw.py

```diff
@@ -60,11 +60,22 @@ def process_spider_exception(_failure):
             return _failure
 
         def process_spider_output(result):
+            def wrapper(result_iterable):
+                try:
+                    for r in result_iterable:
+                        yield r
+                except Exception as ex:
+                    exception_result = process_spider_exception(Failure(ex))
+                    if exception_result is None or isinstance(exception_result, Failure):
+                        raise
+                    for output in exception_result:
+                        yield output
             for method in self.methods['process_spider_output']:
                 result = method(response=response, result=result, spider=spider)
-                assert _isiterable(result), \
-                    'Middleware %s must returns an iterable object, got %s ' % \
-                    (fname(method), type(result))
+                if _isiterable(result):
```

Comment:
@kmike: At first I tried to find a way of checking the result is iterable without exposing the `AssertionError` exception to be catched by some middleware's `process_spider_exception`, but then I thought, since any exception from `process_spider_input` should be passed to `process_spider_exception` (from the docs), it wouldn't be wrong if the result of `process_spider_output` passed too.

That being said, I'm getting a bit confused on what the desired/excepted behavior is/should be, please help me :-)